src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -182,13 +191,22 @@ export const useSwapper = () => {
   }, [])
 
   const sellAssetBalance = useAppSelector(state =>
-    selectPortfolioCryptoBalanceByAssetId(state, { assetId: sellTradeAsset?.asset?.assetId ?? '' }),
-  )
-
-  const feeAsset = useAppSelector(state =>
-    selectFeeAssetById(state, sellTradeAsset?.asset?.assetId ?? 'eip155:1/slip44:60'),
+    selectPortfolioCryptoBalanceByAssetId(state, {
+      assetId: sellTradeAsset?.asset?.assetId ?? '',
+    }),
   )
 
+  const osmosisAsset = useAppSelector(state => selectAssetById(state, osmosisAssetId))
+  const feeAsset = useAppSelector(state => {
+    if (!sellTradeAsset?.asset) {
+      return selectFeeAssetById(state, 'eip155:1/slip44:60')
+    } else if (
+      sellTradeAsset.asset.chainId === KnownChainIds.OsmosisMainnet ||
+      sellTradeAsset.asset.chainId === KnownChainIds.CosmosMainnet
+    ) {
+      return selectFeeAssetById(state, osmosisAsset?.assetId)
+    } else return selectFeeAssetById(state, sellTradeAsset.asset.assetId)
+  })
```

Comment:
this is pretty convoluted - are you trying to get the fee asset from the sell asset? there's a way to get a fee asset from any given asset

Reply:
how do I get the fee asset of ATOM to be Osmosis?

Reply:
This is a weird problem that the platform doesnt currently handle. see my comment here

https://github.com/shapeshift/web/pull/2128#discussion_r923804193